jsons/serializers/default_object.py

```diff
@@ -40,69 +48,189 @@ def default_object_serializer(
     dict will not contain attributes with
     :param verbose: if ``True`` the resulting dict will contain meta
     information (e.g. on how to deserialize).
+    :param strict: a bool to determine if the serializer should be strict
+    (i.e. only dumping stuff that is known to ``cls``).
+    :param fork_inst: if given, it uses this fork of ``JsonSerializable``.
     :param kwargs: any keyword arguments that are to be passed to the
     serializer functions.
     :return: a Python dict holding the values of ``obj``.
     """
     if obj is None:
         return obj
-    strip_attr = strip_attr or []
-    if (not isinstance(strip_attr, MutableSequence)
-            and not isinstance(strip_attr, tuple)):
-        strip_attr = (strip_attr,)
-    cls = cls or obj.__class__
-    obj_dict = _get_dict_from_obj(obj, cls, strip_privates, strip_properties,
-                                  strip_class_variables, strip_attr, **kwargs)
-    kwargs_ = {**kwargs, 'verbose': verbose}
+    _check_slots(cls, fork_inst)
+    strip_attr = _normalize_strip_attr(strip_attr)
+    if strict and cls:
+        attributes = _get_attributes_from_class(
+            cls, strip_privates, strip_properties, strip_class_variables,
+            strip_attr)
+    else:
+        attributes = _get_attributes_from_object(
+            obj, strip_privates, strip_properties, strip_class_variables,
+            strip_attr)
+        cls = obj.__class__
     verbose = Verbosity.from_value(verbose)
-    if Verbosity.WITH_CLASS_INFO in verbose:
-        # Set a flag in kwargs to temporarily store -cls.
-        kwargs_['_store_cls'] = True
-    result = default_dict_serializer(
-        obj_dict,
-        cls,
-        key_transformer=key_transformer,
-        strip_nulls=strip_nulls,
-        strip_privates=strip_privates,
-        strip_properties=strip_properties,
-        strip_class_variables=strip_class_variables,
-        strip_attr=strip_attr,
-        **kwargs_)
-    cls_name = get_class_name(cls, fully_qualified=True,
-                              fork_inst=kwargs['fork_inst'])
+    kwargs_ = {
+        **kwargs,
+        'fork_inst': fork_inst,
+        'verbose': verbose,
+        'strict': strict,
+        # Set a flag in kwargs to temporarily store -cls:
+        '_store_cls': Verbosity.WITH_CLASS_INFO in verbose
+    }
+
+    result = _do_serialize(obj=obj,
+                           cls=cls,
+                           attributes=attributes,
+                           kwargs=kwargs_,
+                           key_transformer=key_transformer,
+                           strip_nulls=strip_nulls,
+                           strip_privates=strip_privates,
+                           strip_properties=strip_properties,
+                           strip_class_variables=strip_class_variables,
+                           strip_attr=strip_attr,
+                           strict=strict,
+                           fork_inst=fork_inst)
```

Comment:
Instead of passing all these variables to every function call, is it possible to keep a global state? Not for this PR, I am just curious about what you think about it.